fuel-core/src/service/modules.rs

```diff
@@ -57,106 +63,121 @@ pub async fn start_modules(config: &Config, database: &Database) -> Result<Modul
-        .import_block_event(block_importer.subscribe())
-        .private_key(
-            hex::decode(
-                "c6bd905dcac2a0b1c43f574ab6933df14d7ceee0194902bce523ed054e8e798b",
-            )
-            .unwrap(),
-        );
+    let relayer = {
+        let mut relayer_builder = fuel_relayer::ServiceBuilder::new();
+        relayer_builder
+            .config(config.relayer.clone())
+            .db(Box::new(database.clone()) as Box<dyn RelayerDb>)
+            .import_block_event(block_importer.subscribe())
+            .private_key(
+                hex::decode(
+                    "c6bd905dcac2a0b1c43f574ab6933df14d7ceee0194902bce523ed054e8e798b",
+                )
+                .unwrap(),
+            );
+
+        relayer_builder.build()?
+    };
 
     let relayer_sender = {
         #[cfg(feature = "relayer")]
         {
-            relayer_builder.sender().clone()
+            relayer.sender().clone()
         }
         #[cfg(not(feature = "relayer"))]
         {
             fuel_core_interfaces::relayer::Sender::noop()
         }
     };
 
-    txpool_builder
-        .config(config.txpool.clone())
-        .db(Box::new(database.clone()) as Box<dyn TxPoolDb>)
-        .import_block_event(block_importer.subscribe());
+    let (tx_status_sender, mut tx_status_receiver) = broadcast::channel(100);
+
+    // Remove once tx_status events are used
+    tokio::spawn(async move { while (tx_status_receiver.recv().await).is_ok() {} });
+
+    let (txpool_sender, txpool_receiver) = mpsc::channel(100);
+    let (incoming_tx_sender, incoming_tx_receiver) = broadcast::channel(100);
 
     #[cfg(feature = "p2p")]
-    let (tx_request_event, rx_request_event) = mpsc::channel(100);
+    let (p2p_request_event_sender, p2p_request_event_receiver) = mpsc::channel(100);
     #[cfg(feature = "p2p")]
-    let (tx_block, rx_block) = mpsc::channel(100);
+    let (block_event_sender, block_event_receiver) = mpsc::channel(100);
 
     #[cfg(not(feature = "p2p"))]
-    let (tx_request_event, _) = mpsc::channel(100);
+    let (p2p_request_event_sender, _p2p_request_event_receiver) = mpsc::channel(100);
     #[cfg(not(feature = "p2p"))]
-    let (_, rx_block) = mpsc::channel(100);
+    let (_block_event_sender, block_event_receiver) = mpsc::channel(100);
 
-    block_importer.start().await;
+    #[cfg(feature = "p2p")]
+    let network_service = {
+        let p2p_db: Arc<dyn P2pDb> = Arc::new(database.clone());
+        let (tx_consensus, _) = mpsc::channel(100);
+        fuel_p2p::orchestrator::Service::new(
+            config.p2p.clone(),
+            p2p_db,
+            p2p_request_event_sender.clone(),
+            p2p_request_event_receiver,
+            tx_consensus,
+            incoming_tx_sender,
+            block_event_sender,
+        )
+    };
+    #[cfg(not(feature = "p2p"))]
+    {
+        let keep_alive = Box::new(incoming_tx_sender);
+        Box::leak(keep_alive);
```

Comment:
Do we also need to leak `block_event_sender` here? I guess the receiver isn't used yet, but it would help future proof the fuel-sync task.

Reply:
Yea, we can certainly do that for `block_event_sender` and any other p2p channel where the receiver does not have owned senders.

Reply:
Updated